tests/backup/backup_test.go

```diff
@@ -131,6 +150,121 @@ var _ = Describe("{BackupCreateRestore}", func() {
 	})
 })
 
+// This test performs basic test of starting an application, backing it up and killing stork while
+// performing backup.
+var _ = Describe("{BackupCreateKillStoreRestore}", func() {
+	var contexts []*scheduler.Context
+	var bkpNamespaces []string
+
+	namespaceMapping := make(map[string]string)
+	labelSelectores := make(map[string]string)
+
+	It("has to connect and check the backup setup", func() {
+		Step("Deploy applications", func() {
+			contexts = make([]*scheduler.Context, 0)
+			bkpNamespaces = make([]string, 0)
+			for i := 0; i < Inst().ScaleFactor; i++ {
+				taskName := fmt.Sprintf("backupcreaterestore-%d", i)
+				appContexts := ScheduleApplications(taskName)
+				contexts = append(contexts, appContexts...)
+				for _, ctx := range appContexts {
+					namespace := GetAppNamespace(ctx, taskName)
+					bkpNamespaces = append(bkpNamespaces, namespace)
+					namespaceMapping[namespace] = fmt.Sprintf("%s-restore", namespace)
+				}
+			}
+			ValidateApplications(contexts)
+		})
+
+		// Wait for IO to run
+		time.Sleep(time.Minute * 20)
+
+		Step(fmt.Sprintf("Create Backup [%s]", BackupName), func() {
+			// TODO(stgleb): Add multi-namespace backup when ready in px-backup
+			for _, namespace := range bkpNamespaces {
+				CreateBackup(fmt.Sprintf("%s-%s", BackupName, namespace),
+					SourceClusterName, BLocationName,
+					[]string{namespace}, labelSelectores, orgID)
+			}
+		})
+
+		Step("Kill stork", func() {
+			timeout := time.Duration(rand.Int()%120)*time.Second + 60
+			time.Sleep(time.Duration(timeout) * time.Second)
+
+			t := func(interval sched.Interval) {
+				ctx := &scheduler.Context{
+					App: &spec.AppSpec{
+						SpecList: []interface{}{
+							&appsapi.Deployment{
+								ObjectMeta: meta_v1.ObjectMeta{
+									Name:      storkDeploymentName,
+									Namespace: storkDeploymentNamespace,
+								},
+							},
+						},
+					},
+				}
+
+				err := Inst().S.DeleteTasks(ctx, nil)
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			taskID, err := sched.Instance().Schedule(t,
+				sched.Periodic(time.Second),
+				time.Now().Add(timeout), true)
+			Expect(err).NotTo(HaveOccurred())
+			defer sched.Instance().Cancel(taskID)
+		})
+
+		Step(fmt.Sprintf("Wait for Backup [%s] to complete", BackupName), func() {
+			for _, namespace := range bkpNamespaces {
+				err := Inst().Backup.WaitForBackupCompletion(
```

Comment:
is it possible that the backups are already completed before we kill stork. If I understand correctly, after killing stork, we expect either backups to fail or pending. Let me know if my understanding is right about this test.

Reply:
well, you've got it. I was also wondering about that. Do you think is there anyother way to syncronize stork killing and backup?

Reply:
Use trigger options in the DeleteTasks API. Supply a trigger function such that it will trigger as soon as your backup starts.

Reply:
Thank you @harsh-px , I'll check it out